mtools/mplotqueries/mplotqueries.py

```diff
@@ -645,13 +645,17 @@ def plot(self):
 
         handles, labels = axis.get_legend_handles_labels()
         if len(labels) > 0:
+            # escape dollar signs in legend labels
+            labels = [x.replace('$', '\$') for x in labels]
```

Comment:
Here we're fetching labels from the axis, escaping any `$` then re-applying the labels to the legend. Instead, could we escape where  plot_instances was enumerated ? ( i think around line 607)